src/battery-check/battery-check.c

```diff
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier: LGPL-1.1-or-later */
+
+#include "log.h"
+#include "main-func.h"
+#include "sleep-config.h"
+
+static int run(int argc, char *argv[]) {
+    int r;
+
+    log_parse_environment();
+    log_open();
+
+    r = battery_is_discharging_and_low();
+    if (r < 0)
+        return log_error_errno(r, "Failed to check battery status: %m");
```

Comment:
This will result in the machine shutting down when it can't read the battery, which is definitely wrong. Log an error, but exit with 0.

Reply:
Awesome sir.
Sometimes I just try to push regardless so I can learn.

Thanks for always coming through sir